volume/fs.go

```diff
@@ -0,0 +1,89 @@
+package volume
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/puppetlabs/wash/activity"
+	"github.com/puppetlabs/wash/plugin"
+)
+
+// FS presents a view of the filesystem of an Execable resource.
+type FS struct {
+	plugin.EntryBase
+	executor plugin.Execable
+}
+
+// NewFS creates a new FS entry with the given name, using the supplied executor to satisfy volume
+// operations.
+func NewFS(name string, executor plugin.Execable) *FS {
+	fs := &FS{EntryBase: plugin.NewEntry(name), executor: executor}
+	// Caching handled in List.
+	fs.DisableCachingFor(plugin.ListOp)
+	return fs
+}
+
+// List will attempt to list the filesystem of an Execable resource (the executor). It will list
+// a directory tree based on supplied configuration (defaulting to `/var/log` if not specified).
+func (d *FS) List(ctx context.Context) ([]plugin.Entry, error) {
+	return List(ctx, d, "")
+}
+
+func consumeExec(ctx context.Context, result plugin.ExecResult) (*bytes.Buffer, error) {
+	var buf bytes.Buffer
+	for chunk := range result.OutputCh {
+		if chunk.Err != nil {
+			activity.Record(ctx, "Error on exec: %v", chunk.Err)
```

Comment:
Hmm, this code implies that `consumeExec` will not return an error even if the command executed successfully, but something went wrong transmitting its output over the network (i.e. ExitCodeCB returning a zero exit code despite a streaming error). Is that intentional and if so, why?

Reply:
I guess it makes sense to return that error. I wasn't sure the error would be fatal enough to avoid returning partial information.